sku.0/sys.server/compiled/game/script/gm/cmd.java

```diff
@@ -654,86 +654,149 @@ else if (faction.equalsIgnoreCase("rebel"))
-            String prompt = "Select the desired rank to set the player to.\n\n";
-            prompt += "Target = (" + target + ") " + getName(target) + "\n";
-            prompt += "Current Rank = (" + rank + ") " + getString(new string_id("faction_recruiter", factions.getRankName(rank, faction)));
-            Vector entries = new Vector();
-            entries.setSize(0);
-            for (int i = 0; i <= factions.MAXIMUM_RANK; i++)
-            {
-                entries = utils.addElement(entries, "(" + i + ") " + getString(new string_id("faction_recruiter", factions.getRankName(i, faction))));
-            }
-            int pid = sui.listbox(self, self, prompt, sui.OK_CANCEL, title, entries, "handleSetRankSelection");
-            if (pid > -1)
-            {
-                utils.setScriptVar(self, "setRank.pid", pid);
-                utils.setScriptVar(self, "setRank.target", target);
-                gm.attachHandlerScript(self);
-            }
+        if (currentRank == goalRank) {
+            sendSystemMessageTestingOnly(self, "The desired rank you specified is already the current rank of "+getPlayerName(target));
+            return SCRIPT_CONTINUE;
         }
-        int newrank = utils.stringToInt(params);
-        if (newrank < 0 || newrank > factions.MAXIMUM_RANK)
-        {
-            sendSystemMessageTestingOnly(self, "/setRank: unable to parse rank. Please choose a rank between 0 & " + factions.MAXIMUM_RANK);
+        if (faction == 0) {
+            sendSystemMessageTestingOnly(self, getPlayerName(target)+" is currently neutral and not aligned with a faction so you cannot set their rank.");
             return SCRIPT_CONTINUE;
         }
-        if (factions.setRank(target, newrank))
-        {
-            int urank = pvpGetCurrentGcwRank(target);
-            sendSystemMessageTestingOnly(self, "(" + target + ") " + getName(target) + "'s rank updated to (" + urank + ") " + getString(new string_id("faction_recruiter", factions.getRankName(urank, faction))));
+
+        int neededRankChange = goalRank - currentRank;
+
+        if (neededRankChange > 0) {
+            do {
+                gcw.increaseGcwRatingToNextRank(target);
+                currentRank = pvpGetCurrentGcwRank(target);
+                command = "increase";
+            } while (currentRank != goalRank);
+        } else if (neededRankChange < 0) {
+            do {
+                gcw.decreaseGcwRatingToPreviousRank(target);
+                currentRank = pvpGetCurrentGcwRank(target);
+                command = "decrease";
+            } while (currentRank != goalRank);
         }
-        else
-        {
-            sendSystemMessageTestingOnly(self, "The system was unable to update (" + target + ") " + getName(target) + "'s rank updated to (" + newrank + ") " + getString(new string_id("faction_recruiter", factions.getRankName(newrank, faction))));
+
+		// it's probably better to read a string file to get these but who has time for that

```

Comment:
Side note... would be more effective to use a switch statement, but this is likely not an issue considering the lower usage.